### PR TITLE
feat(fr_conn): add parser of FriendRequests

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: src/toxcore/friend_connection/packet/friend_requests.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+use toxcore::toxid::NoSpam;
+
+const ONION_SEND_BASE: usize =  PUBLICKEYBYTES + SIZE_IPPORT + MACBYTES;
+const ONION_SEND_1: usize = secretbox::NONCEBYTES + ONION_SEND_BASE * 3;
+const MAX_ONION_DATA_SIZE: usize = ONION_MAX_PACKET_SIZE - (ONION_SEND_1 + 1); // 1 is for packet_id
+const MIN_ONION_DATA_REQUEST_SIZE: usize = 1 + PUBLICKEYBYTES + secretbox::NONCEBYTES + PUBLICKEYBYTES + MACBYTES; // 1 is for packet_id
+/// Maximum size in butes of Onion Data Request packet
+const MAX_DATA_REQUEST_SIZE: usize = MAX_ONION_DATA_SIZE - MIN_ONION_DATA_REQUEST_SIZE;
+/// Minimum size in bytes of Onion Data Response packet
+const MIN_ONION_DATA_RESPONSE_SIZE: usize = PUBLICKEYBYTES + MACBYTES;
```
Comment (kurnevsky) [inline src/toxcore/friend_connection/packet/friend_requests.rs]: Are you sure that this value is correct? It seems nonce is missing here.

Comment (NamsooCho) [reply src/toxcore/friend_connection/packet/friend_requests.rs]: done

**Comment by kpp** [comment]
Where is the definition of FriendRequests in C code?

**Comment by NamsooCho** [comment]
https://github.com/TokTok/c-toxcore/blob/6872c14e1a02445d945623ee6e85230c5d7ecbce/toxcore/friend_connection.c#L821-L844

**Comment by NamsooCho** [comment]
https://github.com/TokTok/c-toxcore/blob/6872c14e1a02445d945623ee6e85230c5d7ecbce/toxcore/Messenger.c#L2477

**Comment by NamsooCho** [comment]
There is no explicit definition of FriendRequests in C-toxcore

**Comment by kpp** [comment]
Thanks

**Pull request merged**